server/src/routes/upload/parseBeatmap.ts

```diff
@@ -261,3 +271,40 @@ const inspectFile = async (file: JSZip.JSZipObject) => {
     }
   }
 }
+
+const getDurationInSeconds = async (
+  sourceFilename: string = 'tempUpload.ogg',
+  audioBuffer: Buffer
+): Promise<number> => {
+  const args = [
+    '-v',
+    'error',
+    '-select_streams',
+    'a:0',
+    '-show_entries',
+    'stream=duration',
+    '-of',
+    'default=noprint_wrappers=1:nokey=1',
+  ]
+
+  // Intermediate file is necessary since an ogg file can't be piped to ffprobe via stdin
+  //   https://stackoverflow.com/questions/3713148/cant-stream-ogg-from-ffmpeg-through-stdout
+  const intermediateFilename = `./${sourceFilename}`
```

Comment:
This should be written to the system's temporary directory, ideally with the `hash` as a subdirectory to avoid race conditions with the same `song.egg`

Reply:
I decided to use `tmp-promise` package for this.  It doesn't add any extra dependencies that the server doesn't already have, and it provides uniqueness and file cleanup.